tests/unit/lms/services/user_test.py

```diff
@@ -38,6 +40,18 @@ def test_upsert_user_with_an_existing_user(
         assert saved_user.roles == lti_user.roles
         assert user == saved_user
```

Comment:
I think there is some missing coverage here. If you comment out the `self._update_user` inside the existing user block the tests still pass.